sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -120,6 +120,24 @@ object ExternalCatalogUtils {
       new Path(totalPath, nextPartPath)
     }
   }
+
+  /**
+   * partition path created by Hive is lower-case, while Spark SQL will
+   * rename it with the partition name in partitionColumnNames, and this function
+   * return the extra lower-case path created by Hive, and then we can delete it.
+   * e.g. /path/A=1/B=2/C=3 rename to /path/A=4/B=5/C=6, the extra path returned is
+   * /path/a=4, which also include all its' child path, such as /path/a=4/b=2
+   */
+  def getExtraPartPathCreatedByHive(
```

Comment:
this is a hive only feature, why do we put it in `ExternalCatalogUtils`? How about the `object HiveExternalCatalog`?